src/Init/Data/Nat/Power2.lean

```diff
@@ -24,7 +24,7 @@ where
 termination_by go p h => n - p
 decreasing_by simp_wf; apply nextPowerOfTwo_dec <;> assumption
 
-def isPowerOfTwo (n : Nat) := ∃ k, n = 2 ^ k
+def isPowerOfTwo (n : Nat) := ∃ k : Nat, n = 2 ^ k
```

Comment:
:(

Reply:
I hope this regression is still acceptable because preventing this information flow is precisely the issue at hand, it's not a side effect we could fix

Reply:
I believe there's a default instance in mathlib for nat powers that makes this work without modification there. I'll have to check. I'm not sure this instance can be brought to core because there are lots of defeq issues here that have to be solved in the algebra hierarchy.

Reply:
@kmill Good to know. I am hopping you will fix the defeq issues in core when you start at the end of the month :)